### DISCUSSION TITLE
Add option to gzip sitemaps?

### DISCUSSION THREAD

**Discussion by darrenshrwd** [ASP-NET-MVC-Boilerplate/Framework#10] [Q&A]
In the home controller of templates could theoretically comment out and gzip as below, however that would negate the point of caching, so  needs to be implemented and tested in **GetSitemapDocuments()** which is part of framework.

            //return this.Content(content, ContentType.Xml, Encoding.UTF8);
            string zippedContent;
            using (var memStreamIn = new MemoryStream(Encoding.Unicode.GetBytes(content)))
            using (var memStreamOut = new MemoryStream())
            using (var gZipStream = new GZipStream(memStreamOut, CompressionMode.Compress))
            {
                memStreamIn.CopyTo(gZipStream);
                var buffer = Encoding.Unicode.GetBytes(content);
                gZipStream.Write(buffer, 0, buffer.Length);

                memStreamOut.Position = 0;
                using (var sr = new StreamReader(memStreamOut))
                {
                    zippedContent = sr.ReadToEnd();
                }
            }
            return Content(zippedContent, "application/x-gzip", Encoding.UTF8);`
 
You may also want to add "application/x-gzip" to your ContentType enumeration.

**Comment by RehanSaeed** [comment]
ASP.NET Core 1.1 added GZIP compression. The template has been updated to include this.

Related: https://github.com/ASP-NET-MVC-Boilerplate/Templates/issues/124

**Comment by darrenshrwd** [comment]
You link to an issue that indicates you decided to NOT implement this in the ASP.NET Core version. However, I figured from that all I **actually** needed was to add the mimeType in my web.config

      <dynamicTypes>
        <add enabled="true" mimeType="application/xml" />
      </dynamicTypes>

**Comment by RehanSaeed** [comment]
The issue I linked was regarding an optional `/sitemap.xml.gz` URL that you can add. There was not much point when the normal `/sitemap.xml` can also be GZIP'ed.
